pkg/splunk/enterprise/searchheadcluster_test.go

```diff
@@ -666,8 +666,8 @@ func TestAppFrameworkSearchHeadClusterShouldFail(t *testing.T) {
 	client.AddObject(&s3Secret)
 
 	_, err = ApplySearchHeadCluster(client, &cr)
-	if err == nil {
-		t.Errorf("ApplySearchHeadCluster should not be successful")
+	if err != nil {
```

Comment:
So what made this test turn from a negative testcase to a positive?

Reply:
The fact that we will not exit from the reconcile if `GetAppListFromS3Bucket` returns error and just continue with other reconcile work will make this test pass now.